drivers/usb/udc/udc_dwc2.c

```diff
@@ -320,27 +341,43 @@ static int dwc2_tx_fifo_write(const struct device *dev,
 	/* Set number of packets and transfer size */
 	sys_write32((pktcnt << USB_DWC2_DEPTSIZN_PKTCNT_POS) | len, dieptsiz_reg);
 
+	if (priv->bufferdma) {
+		if (!dwc2_dma_buffer_ok_to_use(dev, buf->data, len, cfg->mps)) {
+			/* Cannot continue unless buffer is bounced. Device will
+			 * cease to function. Is fatal error appropriate here?
+			 */
+			return -ENOTSUP;
+		}
+
+		sys_write32((uint32_t)buf->data,
+			    (mem_addr_t)&base->in_ep[ep_idx].diepdma);
+
+		sys_cache_data_flush_and_invd_range(buf->data, len);
```

Comment:
Why also invalidate the block, is it not enough to just flush?

Reply:
Flushing would be enough.